### PR TITLE
Add github actions workflow

### PR DESCRIPTION
**Pull request by yatharthranjan**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            ~/.gradle/caches/modules-2/files-2.1/
+            ~/.gradle/caches/modules-2/metadata-2.96/
+            ~/.gradle/native
+            ~/.gradle/wrapper
+          # Key for restoring and saving the cache
+          key: ${{ runner.os }}-gradle
+
+      # Compile the code
+      - name: Compile code
+        run: ./gradlew assemble
```
Comment (blootsvoets) [inline .github/workflows/main.yml:41]: I'd suggest to add
```gradle
assemble.dependsOn shadowJar
```
to build.gradle, so the last step would not be necessary.

**Review by blootsvoets — APPROVED** [review]
LGTM

**Review by Hsankesara — APPROVED** [review]
LGTM :+1:

**Review by afolarin — APPROVED** [review]
LGTM

**Pull request merged**